### PR TITLE
proxmove: init at 1.2

### PR DESCRIPTION
**Pull request by AngryAnt**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/admin/proxmove/derivation.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ lib
```
Comment (lucasew) [inline pkgs/tools/admin/proxmove/derivation.nix:1]: Why not put this code on default.nix?

File: pkgs/tools/admin/proxmove/derivation.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    rm -R assets
+    rm -R artwork
+  '';
+
+  installCheckPhase = ''
+    $out/bin/${pname} --version
+  '';
+
+  meta = with lib; {
+    description = "";
```
Comment (lucasew) [inline pkgs/tools/admin/proxmove/derivation.nix:33]: Fill out description

**Review by lucasew — CHANGES_REQUESTED** [review]

File: pkgs/tools/admin/proxmove/derivation.nix (hunk 2 above)
Comment (AngryAnt) [reply pkgs/tools/admin/proxmove/derivation.nix:33]: Whoops! That went up a bit fast. Thanks :)

File: pkgs/tools/admin/proxmove/derivation.nix (hunk 1 above)
Comment (AngryAnt) [reply pkgs/tools/admin/proxmove/derivation.nix:1]: Yea I can never figure out which is preferable. I've submitted like that before, but then come across packages which do have the split. Personally I agree on the join, so I'll push that update as well.

File: pkgs/tools/admin/proxmove/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  '';
+
+  installCheckPhase = ''
+    $out/bin/${pname} --version
+  '';
+
+  meta = with lib; {
+    description = "The Proxmox VM migrator: migrates VMs between different Proxmox VE clusters";
+    homepage = "https://github.com/ossobv/proxmove";
+    license = licenses.gpl3;
```
[the remark this answers is not in the document]
Comment (AngryAnt) [reply pkgs/tools/admin/proxmove/default.nix:35]: Nice catch! I was unaware of the Plus variant there :)

File: pkgs/tools/admin/proxmove/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  doInstallCheck = true;
+
+  preBuild = ''
+    rm -R assets
+    rm -R artwork
+  '';
+
+  installCheckPhase = ''
+    $out/bin/${pname} --version
+  '';
```
Comment (mweinelt) [inline pkgs/tools/admin/proxmove/default.nix:30]: Run pre and post hooks here.

File: pkgs/tools/admin/proxmove/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
+{ lib
+, python3
+, fetchFromGitHub
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "proxmove";
+  version = "1.2";
```
Comment (mweinelt) [inline pkgs/tools/admin/proxmove/default.nix:8]: Set an explicit format.

File: pkgs/tools/admin/proxmove/default.nix [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+  src = fetchFromGitHub {
+    owner = "ossobv";
+    repo = "proxmove";
+    rev = "v${version}";
+    sha256 = "sha256-8xzsmQsogoMrdpf8+mVZRWPGQt9BO0dBT0aKt7ygUe4=";
```
Comment (mweinelt) [inline pkgs/tools/admin/proxmove/default.nix:14]:
```suggestion
    hash = "sha256-8xzsmQsogoMrdpf8+mVZRWPGQt9BO0dBT0aKt7ygUe4=";
```

File: pkgs/tools/admin/proxmove/default.nix [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [
+    proxmoxer
+  ];
+
+  doInstallCheck = true;
```
Comment (mweinelt) [inline pkgs/tools/admin/proxmove/default.nix:21]: `doCheck` in `buildPython*` maps to `doInstallCheck` in a standard derivation, and defaults to `true` already.

File: pkgs/tools/admin/proxmove/default.nix [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+  preBuild = ''
+    rm -R assets
+    rm -R artwork
+  '';
+
+  installCheckPhase = ''
```
Comment (mweinelt) [inline pkgs/tools/admin/proxmove/default.nix:28]:
```suggestion
  checkPhase = ''
```

**Review by mweinelt — CHANGES_REQUESTED** [review]

File: pkgs/tools/admin/proxmove/default.nix (hunk 5 above)
Comment (AngryAnt) [reply pkgs/tools/admin/proxmove/default.nix:8]: Am I misremembering that "setuptools" is already default for `format` here? Convention is to explicit it regardless while the later-mentioned `doCheck` should rely on the implicitness of its default `true` value?

Comment (AngryAnt) [reply pkgs/tools/admin/proxmove/default.nix:8]: Ah excellent! Thanks :)

**Pull request merged**